src/Specular/FRP/Replaceable.purs

```diff
@@ -25,3 +32,27 @@ weakDynamic :: forall m a. MonadReplace m => MonadFRP m => WeakDynamic (m a) ->
 weakDynamic dyn = do
   Slot {replace} <- newSlot
   subscribeWeakDyn (\x -> replace x) dyn
+
+whenJustD :: forall m a. MonadReplace m => MonadFRP m => Dynamic (Maybe a) -> (Dynamic a -> m Unit) -> m Unit
```

Comment:
@zyla, this requires some rethinking

Reply:
This is reasonable, but lacks certain properties we'd like, for example:

```
whenJustD (Just <$> d) f = f d
```

This is not true because we break out of the frame when propagating the inner dynamic.
It's _true enough_ if we don't care about coincidences, but such cascading frames can cause nasty performance problems with current internals.

When we go ahead with the new internals, this will certainly require rethinking to make it efficient.